core/src/main/java/com/linecorp/armeria/server/DefaultDependencyInjector.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Supplier;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.MoreObjects;
+
+import com.linecorp.armeria.common.annotation.Nullable;
+
+final class DefaultDependencyInjector implements DependencyInjector {
```

Comment:
Not sure `Default` is the best, but `Manual` doesn't sound right to me? 😅

Reply:
If no `DependencyInjector` is configured, we don't use this `DependencyInjector`. So I thought `Default` could be confusing.

Reply:
Had a chat with the team and decided to keep using `DefaultDependencyInjector` because it's a package-private class.
We are going to rename it when we introduce `DependencyInjectorBuilder` with better APIs later. 😉